### PR TITLE
added implementation of IEnumerable to VehicleStorage

### PR DESCRIPTION
**Pull request by andreykalachev**
Added implementation of the Iterator pattern

### CODE REVIEW

File: TenDesignPatterns/Experiments.cs [hunk 1]
```diff
@@ -56,5 +56,15 @@ public static void GetAllVehiclesFromFile()
             var savedVehicles = VehicleToFileLogger.GetVehicles();
             Print($"you have {savedVehicles?.Count()} vehicles saved");
         }
+
+        public static void PrintAllVehiclesInfo()
+        {
+            Print("\nforeach loop through all vehicles");
```
Comment (dmyhun-andersen) [inline TenDesignPatterns/Experiments.cs]: var vehicleStorage = VehicleStorage.GetInstance();

Print("\nforeach loop through all vehicles");
foreach (var vehicle in vehicleStorage)
{
     Print(vehicle.GetInfo());
}

Comment (dmyhun-andersen) [reply TenDesignPatterns/Experiments.cs]: declaration of variables usually should be on top

File: TenDesignPatterns/Persistence/VehicleStorage.cs [hunk 2]
```diff
@@ -1,9 +1,10 @@
-﻿using System.Collections.Generic;
+﻿using System.Collections;
+using System.Collections.Generic;
 using TenDesignPatterns.Models.Interfaces;
 
 namespace TenDesignPatterns.Persistence
 {
-    public class VehicleStorage
+    public class VehicleStorage : IEnumerable
```
Comment (dmyhun-andersen) [inline TenDesignPatterns/Persistence/VehicleStorage.cs]: try IEnumerable<T> where T : IVehicle 
smth like that 
does in working
check

File: TenDesignPatterns/Persistence/VehicleStorageEnumerator.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            _position++;
+            return _position < _vehicles.Count;
+        }
+
+        public void Reset()
+        {
+            _position = -1;
+        }
+
+        object IEnumerator.Current
```
Comment (dmyhun-andersen) [inline TenDesignPatterns/Persistence/VehicleStorageEnumerator.cs]: access modifier

File: TenDesignPatterns/Program.cs [hunk 4]
```diff
@@ -27,6 +27,12 @@ public static void Main(string[] args)
 
             #endregion
 
+            #region Iterator
+
+            PrintAllVehiclesInfo();
```
Comment (dmyhun-andersen) [inline TenDesignPatterns/Program.cs]: PrintAllVehiclesInfo -> PrintAllVehiclesInfoUsingIteratorPattern

File: TenDesignPatterns/Persistence/VehicleStorageEnumerator.cs (hunk 3 above)
Comment (andreykalachev) [reply TenDesignPatterns/Persistence/VehicleStorageEnumerator.cs]: access modifier is not valid for explicit interface implementation

**Pull request merged**